mypy/semanal.py

```diff
@@ -995,7 +995,7 @@ def setup_class_def_analysis(self, defn: ClassDef) -> None:
             defn.info._fullname = defn.info.name()
         if self.is_func_scope() or self.type:
             kind = MDEF
-            if self.is_func_scope():
+            if self.is_nested_func_scope():
```

Comment:
I am not sure this is the right solution. The nested class is clearly an `MDEF`, but on the other hand all these kinds will be refactored soon, so maybe it is OK. @JukkaL what do you think?

Reply:
It's okay to call these LDEFs, since these have access to the local variables of an enclosing function, which is the most important distinction for classes. Whether it's nested within a class or not is less important.